### ISSUE TITLE
Mysql design

### ISSUE THREAD

**Issue by lluiscab** [biels/Minicat-2.0#3]
For saving all the data of the gametypes, showing status on the Web ui, etc. We need to store the information on a database, here's a suggestion of how to do it.

`We will detail more in the future`

**User information**

| userId | userName | userUuid | userPassword | userJoinDate |
| --- | --- | --- | --- | --- |

**Pvp encounters**
`Web UI display > player1 was killed by player2 on Gamemode, x time ago.`

> Also we can add a killType to know if the player was killed using a bow, a sword...

| encounterId | encounterDate | gameId | playerKiller | playerKilled |
| --- | --- | --- | --- | --- |

**Maps**

| mapId | gameId | gameType | mapName | displayItem |
| --- | --- | --- | --- | --- |

**Game history** 

| mapId | gameId | timeStart | timeEnd | winner | players |
| --- | --- | --- | --- | --- | --- |

**Game history** 
`This table is not 100% done`

| performanceId | serverTps | serverTime | runningGames |
| --- | --- | --- | --- |

**Comment by biels** [comment]
Alright, this is the way to go.
However, there are missing fields / tables on the statistics part that will come from other modules. Also, maybe a game_timestamp approach is better than a pvp_encounter approach for saving game statistics?

**Comment by lluiscab** [comment]
Explain me what game_timestamp will do. And also explaim me what you mean with:

> However, there are missing fields / tables on the statistics part that will come from other modules.

**Comment by biels** [comment]
By tracking statistics I mean having a table `game_timestamps` where every second / every 5 seconds a record is created containing the current game status. Also a `player_game_timestamps` table exists containing the player specific info at every moment. Using this information you could easily display charts of player's damage dealt / taken or a pie chart showing the different items a player had in hand during the game.
Basically, a periodic update tick approach. What you(@lluiscab) proposed is event driven. 
So the thing is: **event driven** vs **tick driven**
Which would you choose and why?

> However, there are missing fields / tables on the statistics part that will come from other modules.

This refers to for example the build of the player (items from the in-game shop that the user bought and at which point in time)

**Comment by lluiscab** [comment]
I think **event driven** is best. Why we need to know second by second what's happening on the game? I think storing this information is not useful. We have to decide what we want to do with the stored information, and then, decide what information we need to store.

**Comment by biels** [comment]
Basically **event driven** is best when there are few events on average and **tick driven** is better when there are so many events that storing them all would be an overkill.
For example, think about damage events or gold earning events. 
I agree that event driven is better and more accurate, but how would you handle those example statistics?
